src/allotropy/parsers/agilent_gen5/agilent_gen5_structure.py

```diff
@@ -268,42 +268,65 @@ def create(cls, reader: LinesReader) -> ReadData:
+        read_sections = []
+        for read_section in section_lines_reader.iter_sections("^Read\t"):
+            read_sections.append(read_section)
+
+        section = 0
+        for read_mode_idx in range(len(read_modes)):
+            read_mode = read_modes[read_mode_idx]
+            while section < len(read_sections):
+                join_read_section_lines = "\n".join(read_sections[section].lines)
+                device_control_data = DeviceControlData.create(join_read_section_lines, read_mode)
+                measurement_labels = cls._get_measurement_labels(device_control_data, read_mode)
+                number_of_averages = device_control_data.get(MEASUREMENTS_DATA_POINT_KEY)
+                read_height = device_control_data.get(READ_HEIGHT_KEY) or ""
+                read_data_list.append(ReadData(
+                    read_mode=read_mode,
+                    step_label=device_control_data.step_label,
+                    measurement_labels=measurement_labels,
+                    detector_carriage_speed=device_control_data.get(READ_SPEED_KEY),
+                    # Absorbance attributes
+                    pathlength_correction=device_control_data.get(PATHLENGTH_CORRECTION_KEY),
+                    number_of_averages=try_float_or_none(number_of_averages),
+                    # Luminescence attributes
+                    detector_distance=try_float_or_none(read_height.split(" ")[0]),
+                    # Fluorescence attributes
+                    filter_sets=cls._get_filter_sets(
+                        measurement_labels, device_control_data, read_mode
+                    ),
+                ))
+                section += 1
+                break
+        return read_data_list
 
     @staticmethod
-    def get_read_mode(procedure_details: str) -> ReadMode:
-        read_modes = [
-            read_mode for read_mode in ReadMode if read_mode.value in procedure_details
-        ]
+    def get_read_modes(procedure_details: str) -> list[ReadMode]:
+        read_modes = []
+        for read_mode in ReadMode:
+            # Construct the regex pattern for the current read mode
+            pattern = r"\t{} Endpoint".format(re.escape(read_mode.value))
+            # Use regex to find all occurrences of the read mode pattern in the procedure details
+            matches = re.findall(pattern, procedure_details)
+            if matches:
+                # Add the read_mode to the list for each match found
+                read_modes.extend([read_mode] * len(matches))
+
         if not read_modes:
             raise AllotropeConversionError(UNSUPPORTED_READ_MODE_ERROR)
-        if len(read_modes) > 1:
-            raise AllotropeConversionError(MULTIPLE_READ_MODE_ERROR)
 
-        if read_modes[0] in (ReadMode.FLUORESCENCE, ReadMode.ALPHALISA):
-            return ReadMode.FLUORESCENCE
-        return read_modes[0]
+        # Replace ALPHALISA with FLUORESCENCE
```

Comment:
Handle case where there is `ALPHALISA` and `FLUORESCENCE`. If we expect it to never happen, check for it and raise an `AllotropeConversionError`